### PR TITLE
Use arm64 version of Z3 4.12.1 for arm64 macOS releases of Dafny

### PR DESCRIPTION
**Pull request by 17451k**
Implements #4304.

This PR is not yet complete as it requires a new snapshot in the https://github.com/dafny-lang/solver-builds repo. @atomb, could you please create it?

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

**Comment by atomb** [comment]
And I just edited the PR to use the new snapshot, so you shouldn't need to do anything. :)

**Review by atomb — APPROVED** [review]
Thanks for doing this, @17451k!

File: Makefile [hunk 1]
```diff
@@ -26,23 +26,34 @@ refman-release: exe
...
+	rm z3-4.8.5-x64-macos-11-bin.zip
+	mv z3-* ${DIR}/Binaries/z3/bin/
+	chmod +x ${DIR}/Binaries/z3/bin/z3-*
+
+z3-mac-arm:
+	mkdir -p ${DIR}Binaries/z3/bin
+	wget https://github.com/dafny-lang/solver-builds/releases/download/snapshot-2023-08-02/z3-4.12.1-arm64-macos-11-bin.zip
+	unzip z3-4.12.1-arm64-macos-11-bin.zip
+	rm z3-4.12.1-arm64-macos-11-bin.zip
+	wget https://github.com/dafny-lang/solver-builds/releases/download/snapshot-2023-08-02/z3-4.8.5-arm64-macos-11-bin.zip
```
Comment (17451k) [inline Makefile:43]: @atomb These 3 lines are not correct, you need to replace z3-4.8.5-arm64 with z3-4.8.5-x64! Not sure how to do it by myself when PR is already merged.

Comment (atomb) [reply Makefile:43]: Oh, good catch! I'll fix it in a separate PR.

**Pull request merged**